File: 1776-minimum-operations-to-reduce-x-to-zero/minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        i = 0
        j = 0
        max_window = - 1
        current_sum = 0
        total_sum = sum(nums)
        target = total_sum - x

        while (j < len(nums)):
            current_sum += nums[j]

            while (current_sum > target and i <= j):
                current_sum -= nums[i]
                i += 1
            
            if (current_sum == target):
                max_window = max(max_window, j -i + 1)
            
            j += 1

        
        return max_window if max_window == - 1 else len(nums) - max_window

File: 1776-minimum-operations-to-reduce-x-to-zero/minimum-operations-to-reduce-x-to-zero.js
var minOperations = function(nums, x) {
    let totalSum = nums.reduce((a, b)=> a + b)
    let target = totalSum - x
    let i = 0
    let j = 0
    let output = - 1
    let currentSum = 0

    while (j < nums.length) {
        currentSum += nums[j]

        while (currentSum > target && i <= j) {
            currentSum -= nums[i]
            i += 1
        }

        if (currentSum == target) {
            output = Math.max(output, j - i + 1)
        }


        j += 1
    }
    return output == -1 ? -1 : nums.length - output
};